File: SkateboardAS/Core/Interfaces/IAgvService.cs
using Core.Models;

namespace Core.Interfaces;

public interface IAgvService
{
    Task<CommandResult> LoadProgramAsync(string programName, CancellationToken ct = default);
    Task<CommandResult> ExecuteProgramAsync(CancellationToken ct = default);
    Task<AgvStatus> GetAgvStatusAsync(CancellationToken ct = default);
}

File: SkateboardAS/core/src/main/java/dk/sdu/core/api/IAgvService.java
package dk.sdu.core.api;

import dk.sdu.core.model.AgvStatus;
import dk.sdu.core.model.CommandResult;

public interface IAgvService {
    CommandResult loadProgram(String programName);
    CommandResult executeProgram();
    AgvStatus getAgvStatus();
}
